### PR TITLE
Single Precision maths for RSA, DH and ECC

### PR DESCRIPTION
**Pull request by SparkiDev**
Single Precision maths for RSA (and DH)
Single Precision ECC implementation
Code generated using ruby script.

### CODE REVIEW

**Comment by toddouska** [comment]
The speedups themselves are working great, thanks.

File: certs/client-cert-3072.pem [hunk 1]
```diff
@@ -0,0 +1,108 @@
+Certificate:
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply certs/client-cert-3072.pem]: Fixed and added test case.

File: certs/client-key-3072.pem [hunk 2]
```diff
@@ -0,0 +1,40 @@
+-----BEGIN PRIVATE KEY-----
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply certs/client-key-3072.pem]: Fixed and added test case that uses the cert and key.

File: configure.ac [hunk 3]
```diff
@@ -3182,6 +3189,102 @@ AC_ARG_WITH([intelqa],
 
 AM_CONDITIONAL([BUILD_INTEL_QA], [test "x$ENABLED_INTEL_QA" = "xyes"])
 
+# Single Precision maths implementation
+AC_ARG_ENABLE([sp],
+    [AS_HELP_STRING([--enable-sp],[Enable Single Precision maths implementation (default: disabled)])],
+    [ ENABLED_SP=$enableval ],
+    [ ENABLED_SP=yes ],
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply configure.ac]: Do you want 2048, 3072 and p256 on by default?
I had to turn it on for testing purposes but I can change this now.

[the remark this answers is not in the document]
Comment (toddouska) [reply configure.ac]: --enable-sp should default to off since it's new.

If someone enables sp all the common sizes you implemented should be on.

**Pull request merged**